packages/flutter/lib/src/rendering/sliver_group.dart

```diff
@@ -0,0 +1,146 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:vector_math/vector_math_64.dart';
+
+import 'object.dart';
+import 'sliver.dart';
+
+/// A sliver that places multiple sliver children in a linear array along the cross
+/// axis.
+///
+/// Since the extent of the viewport in the cross axis direction is finite,
+/// each sliver is forced to divide up the extent per specification via
+/// [ParentDataWidget] [SliverConstrainedCrossAxis] and [SliverCrossAxisExpanded].
+///
+/// The algorithm for dividing up the cross axis extent is as follows.
+/// Every widget either has constrained cross axis extent or a flex value.
+/// First allocate space to all the widgets with constrained cross axis extent.
+/// Then divide up the remaining space proportional to the widget's flex value.
+/// Flex values can be specified via the [SliverCrossAxisExpanded] widgets, but if
+/// a sliver without [SliverCrossAxisExpanded] or [SliverConstrainedCrossAxis] is passed in,
+/// it is assumed to have a flex value of 1, which by itself expands
+/// to take up all of the cross axis extent.
+class RenderSliverCrossAxisGroup extends RenderSliver with ContainerRenderObjectMixin<RenderSliver, SliverPhysicalContainerParentData> {
```

Comment:
re: your comment about this being the same as RenderProxySliver as far as applyPaintTransform, then should that be the super class? RenderProxySliver only has one child though...

Reply:
So it seems `RenderBox` has a default `applyPaintTransform` already implemented in the same way `RenderProxySliver.applyPaintTransform` and my new `SliverCrossAxisGroup.applyPaintTransform`. Perhaps this is a refactor that can be done in a separate PR?

Reply:
For reference, in `box.dart`, we have
https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/rendering/box.dart#L2539-L2563
whereas in `sliver.dart`, we have
https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/rendering/sliver.dart#L1434-L1439

Reply:
> Perhaps this is a refactor that can be done in a separate PR?

If we know it should be different, we should not land it and then change it in a separate PR.

Reply:
Figured out how to test the `applyPaintTransform` with @HansMuller's help.